File: myinfarith/MyInfArith.java
package myinfarith;

import arbitraryarithmetic.*;

public class MyInfArith {
    public static void main(String[] args) {
        if(args.length != 4) {
            System.out.println("Usage: java MyInfArith int/float add/sub/mul/div <number1> <number2>");
            return;
        }
        String type = args[0];
        String operation = args[1];
        String num1 = args[2];
        String num2 = args[3];
        if(type=="int"){
            AInteger first = new AInteger(num1);
            AInteger second = new AInteger(num2);
            switch(operation) {
                case "add":
                    System.out.println(AInteger.add(first, second).value);
                    break;
                case "sub":
                    System.out.println(AInteger.sub(first, second).value);
                    break;
                case "mul":
                    System.out.println(AInteger.mul(first, second).value);
                    break;
                case "div":
                    System.out.println(AInteger.div(first, second).value);
                    break;
                default:
                    System.out.println("Invalid operation. Use add, sub, mul, or div.");
                    break;    
            }
        }
        else if(type=="float"){
            AFloat first = new AFloat(num1);
            AFloat second = new AFloat(num2);
            switch(operation) {
                case "add":
                    System.out.println(AFloat.add(first, second).value);
                    break;
                case "sub":
                    System.out.println(AFloat.sub(first, second).value);
                    break;
                case "mul":
                    System.out.println(AFloat.mul(first, second).value);
                    break;
                case "div":
                    System.out.println(AFloat.div(first, second).value);
                    break;
                default:
                    System.out.println("Invalid operation. Use add, sub, mul, or div.");
                    break;    
            }
        }
        else{
            System.out.println("Invalid type. Use int or float.");
        }
    }
}

File: MyInfArith.py
# import subprocess
# import os
# import sys

# class MyInfArith :
#     def __init__(self,mode):
#         self.file_path="C:/Projects/SDF_project_CS58/src/main/java/"
#         self.mode=mode

#     def compile(self):
#         os.system(f"javac -cp .:{self.file_path}target/aarithmetic.jar {self.file_path}arbitraryarithmetic/MyInfArith.java")
    
#     def run(self,operation, num1, num2):
#         os.system(f"java -cp .:{self.file_path}target/aarithmetic.jar {self.file_path}myinfarith.MyInfArith {self.mode} {operation} {num1} {num2}")

# def main():
#     import sys

# def main():
#     if len(sys.argv) != 5:
#         print("Usage: python script.py <mode> <operation> <num1> <num2>")
#         return

#     mode = sys.argv[1]
#     operation = sys.argv[2]
#     num1 = sys.argv[3]
#     num2 = sys.argv[4]

#     app = MyInfArith(mode)
#     app.compile()
#     app.run(operation, num1, num2)

# if __name__ == "__main__":
#     main()
import subprocess
import sys
import os

class MyInfArith:
    def __init__(self, mode):
        self.mode = mode
        self.java_src_path = "C:/Projects/SDF_project_CS58/src/main/java"
        self.classpath = "C:/Projects/SDF_project_CS58/target/classes"
        self.full_classname = "arbitraryarithmetic.MyInfArith" 

    def compile(self):
        compile_cmd = [
            "javac",
            "-d", self.classpath,
            "-cp", self.classpath,
            os.path.join(self.java_src_path, "arbitraryarithmetic", "MyInfArith.java")
        ]
        subprocess.run(compile_cmd, check=True)

    def run(self, operation, num1, num2):
        run_cmd = [
            "java",
            "-cp", self.classpath,
            self.full_classname,
            self.mode, operation, num1, num2
        ]
        subprocess.run(run_cmd, check=True)

def main():
    if len(sys.argv) != 5:
        print("Usage: python MyInfArith"
        ".py <mode> <operation> <num1> <num2>")
        return

    mode = sys.argv[1]
    operation = sys.argv[2]
    num1 = sys.argv[3]
    num2 = sys.argv[4]

    app = MyInfArith(mode)
    #app.compile()
    app.run(operation, num1, num2)

if __name__ == "__main__":
    main()


